ocw/lib/gce.py

```diff
@@ -207,3 +209,112 @@ def cleanup_all(self):
                 if "warnings" in response:
                     for w in response["warnings"]:
                         self.log_warn(w["message"])
+        self.cleanup_vault_keys()
+
+    def cleanup_vault_keys(self):
+        self.log_info("## GCE Vault key cleaning - %s ##" % self.__project)
+        credentials = service_account.Credentials.from_service_account_info(
+            self.__credentials.getPrivateKeyData()
+        )
+        service = googleapiclient.discovery.build('iam', 'v1', credentials=credentials)
+        # now get service accounts
+        vault_emails = self.parser_service_account_list(service)
+        # for each vault account find the associated names with old keys
+        vault_keys = self.get_vault_name(service, vault_emails)
+
+        self.delete_vault_keys(service, vault_keys)
+
+    @filterAfterTime(hours=24)
+    def get_vault_name(self, service, vault_account_tuple):
+        ''' Returns the names of the Service Accounts filtered by date
+
+        Retieve the Service Accounts with the keys which have expired.
+        The expiration date is set to 24 hours.
+
+        Parameters
+        ----------
+        service : googleapiclient.discovery.Resource, required
+            The IAM Service Account instance
+        vault_account_tuple : iterable, required
+            An iterable object with strings of the Email address associated to the service accounts.
+        '''
+        keys = []
+        for email in vault_account_tuple:
+            service_resp = service.projects().serviceAccounts().keys().list(
+                name='projects/-/serviceAccounts/%s' % email).execute()
+            keys += [k for k in service_resp['keys']]
+        return keys
+
+    @filterService(name='vaultopenqa')
```

Comment:
Is this decorator configurable by the caller of `paresr_service_account_list()` ?

Reply:
it can become obviously, but i am not sure what do you have in mind exactly.